pyhealth/unittests/test_datasets/test_mimic4

```diff
@@ -0,0 +1,96 @@
+import datetime
+import unittest
+
+from pyhealth.datasets import MIMIC4Dataset
+
+# this test suite verifies the MIMIC4 demo dataset is parsed correctly and produces
+# the correct dataset for demoing. To qualify the units under test we check the dataset statistics, 
+# and a single sample from the dataset.
+# 
+# the dataset can be found at https://physionet.org/content/mimic-iv-demo/2.2/
+# this test may because
+# a. MIMIC4, or BaseDataset source code was modified, introducing a bug
+# b. the dataset available at the file root has been modified
```

Comment:
let me know if this test approach is acceptable for you. I don't really like that we have to rely on the dataset remaining the same, but given that this class is all about providing test data that makes development easy, it makes sense to check that the data we provide is consistent. Otherwise the data scientist using pyhealth may come to find their models have started generating new predictions/results and old algorithms are no longer effective.